### PR TITLE
Feat: Add to cart functionality

### PR DESCRIPTION
**Pull request by deepak29-git**
1.Added add to cart feature
2.added auth feature in login and log out

### CODE REVIEW

File: src/Context/auth-context.js [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+            lastName: user.lastName,
+            email: user.email,
+            newPassword: user.newPassword,
+            confirmPassword:user.confirmPassword
+        })
+        localStorage.setItem('user',JSON.stringify(data))
+        const {foundUser,encodedToken}=data
+        authDispatch({type:"SET_TOKEN",payload:encodedToken})
+        authDispatch({type:"SET_USER",payload:foundUser})
+        localStorage.setItem("user",JSON.stringify({
```
Comment (tanishk15sharma) [inline src/Context/auth-context.js:53]: Dont need to put data in local Storage , just put the token

File: src/Pages/Cart/Cart.jsx [hunk 2]
```diff
@@ -1,3 +1,104 @@
...
+
+  const user = JSON.parse(localStorage.getItem("user"));
+  useEffect(() => {
+    (async () => {
+      const { data } = await axios.get("/api/user/cart", {
+        headers: {
+          authorization: user.token, // passing token as an authorization header
+        },
+      });
+      console.log(data);
```
Comment (tanishk15sharma) [inline src/Pages/Cart/Cart.jsx:20]: remove console from here

**Review by tanishk15sharma — APPROVED** [review]
It is not need to put user data in local storage you can change it , else looks good to me

File: src/Context/auth-context.js (hunk 1 above)
Comment (deepak29-git) [reply src/Context/auth-context.js:53]: ok

File: src/Pages/Cart/Cart.jsx (hunk 2 above)
Comment (deepak29-git) [reply src/Pages/Cart/Cart.jsx:20]: ok

File: src/Pages/Cart/Cart.jsx [hunk 3]
```diff
@@ -1,3 +1,104 @@
-export const Cart=()=>{
-    return <h1>Cart</h1>
-}
\ No newline at end of file
+import axios from "axios";
+import { useEffect } from "react";
+import { Header } from "../../components/Header/Header";
+import { useCart } from "../../Context/cart-context";
+// import { addToCart } from "../../Utilities/add-to-cart";
```
Comment (Mohitkumar33) [inline src/Pages/Cart/Cart.jsx:5]: remove the commented code

**Review by Mohitkumar33 — APPROVED** [review]
nice code

**Pull request merged**